#!/bin/bash

# Move to the directory where the BAM and GFF data are stored:
cd /"..."

# Count all reads in CDS:
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=CDS --idattr=Parent --mode=union ./BAM_sorted/28cell1s.bam ./features/exon.gff > 28cell1cds.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=CDS --idattr=Parent --mode=union ./BAM_sorted/28cell2s.bam ./features/exon.gff > 28cell2cds.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=CDS --idattr=Parent --mode=union ./BAM_sorted/28cell3s.bam ./features/exon.gff > 28cell3cds.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=CDS --idattr=Parent --mode=union ./BAM_sorted/blastula2s.bam ./features/exon.gff > blastula2cds.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=CDS --idattr=Parent --mode=union ./BAM_sorted/blastula3s.bam ./features/exon.gff > blastula3cds.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=CDS --idattr=Parent --mode=union ./BAM_sorted/dlarv1s.bam ./features/exon.gff > dlarv1cds.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=CDS --idattr=Parent --mode=union ./BAM_sorted/dlarv2s.bam ./features/exon.gff > dlarv2cds.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=CDS --idattr=Parent --mode=union ./BAM_sorted/dlarv3s.bam ./features/exon.gff > dlarv3cds.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=CDS --idattr=Parent --mode=union ./BAM_sorted/gastrula1s.bam ./features/exon.gff > gastrula1cds.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=CDS --idattr=Parent --mode=union ./BAM_sorted/gastrula2s.bam ./features/exon.gff > gastrula2cds.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=CDS --idattr=Parent --mode=union ./BAM_sorted/gastrula3s.bam ./features/exon.gff > gastrula3cds.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=CDS --idattr=Parent --mode=union ./BAM_sorted/morula1s.bam ./features/exon.gff > morula1cds.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=CDS --idattr=Parent --mode=union ./BAM_sorted/morula2s.bam ./features/exon.gff > morula2cds.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=CDS --idattr=Parent --mode=union ./BAM_sorted/ovo1s.bam ./features/exon.gff > ovo1cds.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=CDS --idattr=Parent --mode=union ./BAM_sorted/ovo2s.bam ./features/exon.gff > ovo2cds.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=CDS --idattr=Parent --mode=union ./BAM_sorted/ovo3s.bam ./features/exon.gff > ovo3cds.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=CDS --idattr=Parent --mode=union ./BAM_sorted/spat1s.bam ./features/exon.gff > spat1cds.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=CDS --idattr=Parent --mode=union ./BAM_sorted/spat2s.bam ./features/exon.gff > spat2cds.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=CDS --idattr=Parent --mode=union ./BAM_sorted/spat3s.bam ./features/exon.gff > spat3cds.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=CDS --idattr=Parent --mode=union ./BAM_sorted/troc1s.bam ./features/exon.gff > troc1cds.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=CDS --idattr=Parent --mode=union ./BAM_sorted/troc2s.bam ./features/exon.gff > troc2cds.counts

# Count all reads in genes:
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=mRNA --idattr=ID --mode=union ./BAM_sorted/2nst0_1s.bam ./features/gene.gff > 2nst0_1gen.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=mRNA --idattr=ID --mode=union ./BAM_sorted/2nst0_2s.bam ./features/gene.gff > 2nst0_2gen.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=mRNA --idattr=ID --mode=union ./BAM_sorted/2nst0_3s.bam ./features/gene.gff > 2nst0_3gen.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=mRNA --idattr=ID --mode=union ./BAM_sorted/3nst0_1s.bam ./features/gene.gff > 3nst0_1gen.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=mRNA --idattr=ID --mode=union ./BAM_sorted/3nst0_2s.bam ./features/gene.gff > 3nst0_2gen.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=mRNA --idattr=ID --mode=union ./BAM_sorted/3nst0_3s.bam ./features/gene.gff > 3nst0_3gen.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=mRNA --idattr=ID --mode=union ./BAM_sorted/3nst3fa_1s.bam ./features/gene.gff > 3nst3fa_1gen.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=mRNA --idattr=ID --mode=union ./BAM_sorted/3nst3fa_2s.bam ./features/gene.gff > 3nst3fa_2gen.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=mRNA --idattr=ID --mode=union ./BAM_sorted/3nst3fa_3s.bam ./features/gene.gff > 3nst3fa_3gen.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=mRNA --idattr=ID --mode=union ./BAM_sorted/3nst3fb_1s.bam ./features/gene.gff > 3nst3fb_1gen.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=mRNA --idattr=ID --mode=union ./BAM_sorted/3nst3fb_2s.bam ./features/gene.gff > 3nst3fb_2gen.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=mRNA --idattr=ID --mode=union ./BAM_sorted/3nst3fb_3s.bam ./features/gene.gff > 3nst3fb_3gen.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=mRNA --idattr=ID --mode=union ./BAM_sorted/3nst23ma_1s.bam ./features/gene.gff > 3nst23ma_1gen.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=mRNA --idattr=ID --mode=union ./BAM_sorted/3nst23ma_2s.bam ./features/gene.gff > 3nst23ma_2gen.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=mRNA --idattr=ID --mode=union ./BAM_sorted/3nst23ma_3s.bam ./features/gene.gff > 3nst23ma_3gen.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=mRNA --idattr=ID --mode=union ./BAM_sorted/3nst23mb_1s.bam ./features/gene.gff > 3nst23mb_1gen.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=mRNA --idattr=ID --mode=union ./BAM_sorted/3nst23mb_2s.bam ./features/gene.gff > 3nst23mb_2gen.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=mRNA --idattr=ID --mode=union ./BAM_sorted/3nst23mb_3s.bam ./features/gene.gff > 3nst23mb_3gen.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=mRNA --idattr=ID --mode=union ./BAM_sorted/28cell1s.bam ./features/gene.gff > 28cell1gen.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=mRNA --idattr=ID --mode=union ./BAM_sorted/28cell2s.bam ./features/gene.gff > 28cell2gen.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=mRNA --idattr=ID --mode=union ./BAM_sorted/28cell3s.bam ./features/gene.gff > 28cell3gen.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=mRNA --idattr=ID --mode=union ./BAM_sorted/blastula2s.bam ./features/gene.gff > blastula2gen.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=mRNA --idattr=ID --mode=union ./BAM_sorted/blastula3s.bam ./features/gene.gff > blastula3gen.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=mRNA --idattr=ID --mode=union ./BAM_sorted/dlarv1s.bam ./features/gene.gff > dlarv1gen.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=mRNA --idattr=ID --mode=union ./BAM_sorted/dlarv2s.bam ./features/gene.gff > dlarv2gen.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=mRNA --idattr=ID --mode=union ./BAM_sorted/dlarv3s.bam ./features/gene.gff > dlarv3gen.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=mRNA --idattr=ID --mode=union ./BAM_sorted/gastrula1s.bam ./features/gene.gff > gastrula1gen.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=mRNA --idattr=ID --mode=union ./BAM_sorted/gastrula2s.bam ./features/gene.gff > gastrula2gen.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=mRNA --idattr=ID --mode=union ./BAM_sorted/gastrula3s.bam ./features/gene.gff > gastrula3gen.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=mRNA --idattr=ID --mode=union ./BAM_sorted/morula1s.bam ./features/gene.gff > morula1gen.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=mRNA --idattr=ID --mode=union ./BAM_sorted/morula2s.bam ./features/gene.gff > morula2gen.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=mRNA --idattr=ID --mode=union ./BAM_sorted/ovo1s.bam ./features/gene.gff > ovo1gen.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=mRNA --idattr=ID --mode=union ./BAM_sorted/ovo2s.bam ./features/gene.gff > ovo2gen.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=mRNA --idattr=ID --mode=union ./BAM_sorted/ovo3s.bam ./features/gene.gff > ovo3gen.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=mRNA --idattr=ID --mode=union ./BAM_sorted/spat1s.bam ./features/gene.gff > spat1gen.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=mRNA --idattr=ID --mode=union ./BAM_sorted/spat2s.bam ./features/gene.gff > spat2gen.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=mRNA --idattr=ID --mode=union ./BAM_sorted/spat3s.bam ./features/gene.gff > spat3gen.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=mRNA --idattr=ID --mode=union ./BAM_sorted/troc1s.bam ./features/gene.gff > troc1gen.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=mRNA --idattr=ID --mode=union ./BAM_sorted/troc2s.bam ./features/gene.gff > troc2gen.counts


# Count all reads in introns:
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=intrn --idattr=Parent --mode=union ./BAM_sorted/2nst0_1s.bam ./features/intron.gff > 2nst0_1int.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=intrn --idattr=Parent --mode=union ./BAM_sorted/2nst0_2s.bam ./features/intron.gff > 2nst0_2int.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=intrn --idattr=Parent --mode=union ./BAM_sorted/2nst0_3s.bam ./features/intron.gff > 2nst0_3int.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=intrn --idattr=Parent --mode=union ./BAM_sorted/3nst0_1s.bam ./features/intron.gff > 3nst0_1int.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=intrn --idattr=Parent --mode=union ./BAM_sorted/3nst0_2s.bam ./features/intron.gff > 3nst0_2int.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=intrn --idattr=Parent --mode=union ./BAM_sorted/3nst0_3s.bam ./features/intron.gff > 3nst0_3int.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=intrn --idattr=Parent --mode=union ./BAM_sorted/3nst3fa_1s.bam ./features/intron.gff > 3nst3fa_1int.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=intrn --idattr=Parent --mode=union ./BAM_sorted/3nst3fa_2s.bam ./features/intron.gff > 3nst3fa_2int.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=intrn --idattr=Parent --mode=union ./BAM_sorted/3nst3fa_3s.bam ./features/intron.gff > 3nst3fa_3int.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=intrn --idattr=Parent --mode=union ./BAM_sorted/3nst3fb_1s.bam ./features/intron.gff > 3nst3fb_1int.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=intrn --idattr=Parent --mode=union ./BAM_sorted/3nst3fb_2s.bam ./features/intron.gff > 3nst3fb_2int.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=intrn --idattr=Parent --mode=union ./BAM_sorted/3nst3fb_3s.bam ./features/intron.gff > 3nst3fb_3int.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=intrn --idattr=Parent --mode=union ./BAM_sorted/3nst23ma_1s.bam ./features/intron.gff > 3nst23ma_1int.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=intrn --idattr=Parent --mode=union ./BAM_sorted/3nst23ma_2s.bam ./features/intron.gff > 3nst23ma_2int.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=intrn --idattr=Parent --mode=union ./BAM_sorted/3nst23ma_3s.bam ./features/intron.gff > 3nst23ma_3int.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=intrn --idattr=Parent --mode=union ./BAM_sorted/3nst23mb_1s.bam ./features/intron.gff > 3nst23mb_1int.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=intrn --idattr=Parent --mode=union ./BAM_sorted/3nst23mb_2s.bam ./features/intron.gff > 3nst23mb_2int.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=intrn --idattr=Parent --mode=union ./BAM_sorted/3nst23mb_3s.bam ./features/intron.gff > 3nst23mb_3int.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=intrn --idattr=Parent --mode=union ./BAM_sorted/28cell1s.bam ./features/intron.gff > 28cell1int.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=intrn --idattr=Parent --mode=union ./BAM_sorted/28cell2s.bam ./features/intron.gff > 28cell2int.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=intrn --idattr=Parent --mode=union ./BAM_sorted/28cell3s.bam ./features/intron.gff > 28cell3int.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=intrn --idattr=Parent --mode=union ./BAM_sorted/blastula2s.bam ./features/intron.gff > blastula2int.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=intrn --idattr=Parent --mode=union ./BAM_sorted/blastula3s.bam ./features/intron.gff > blastula3int.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=intrn --idattr=Parent --mode=union ./BAM_sorted/dlarv1s.bam ./features/intron.gff > dlarv1int.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=intrn --idattr=Parent --mode=union ./BAM_sorted/dlarv2s.bam ./features/intron.gff > dlarv2int.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=intrn --idattr=Parent --mode=union ./BAM_sorted/dlarv3s.bam ./features/intron.gff > dlarv3int.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=intrn --idattr=Parent --mode=union ./BAM_sorted/gastrula1s.bam ./features/intron.gff > gastrula1int.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=intrn --idattr=Parent --mode=union ./BAM_sorted/gastrula2s.bam ./features/intron.gff > gastrula2int.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=intrn --idattr=Parent --mode=union ./BAM_sorted/gastrula3s.bam ./features/intron.gff > gastrula3int.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=intrn --idattr=Parent --mode=union ./BAM_sorted/morula1s.bam ./features/intron.gff > morula1int.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=intrn --idattr=Parent --mode=union ./BAM_sorted/morula2s.bam ./features/intron.gff > morula2int.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=intrn --idattr=Parent --mode=union ./BAM_sorted/ovo1s.bam ./features/intron.gff > ovo1int.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=intrn --idattr=Parent --mode=union ./BAM_sorted/ovo2s.bam ./features/intron.gff > ovo2int.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=intrn --idattr=Parent --mode=union ./BAM_sorted/ovo3s.bam ./features/intron.gff > ovo3int.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=intrn --idattr=Parent --mode=union ./BAM_sorted/spat1s.bam ./features/intron.gff > spat1int.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=intrn --idattr=Parent --mode=union ./BAM_sorted/spat2s.bam ./features/intron.gff > spat2int.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=intrn --idattr=Parent --mode=union ./BAM_sorted/spat3s.bam ./features/intron.gff > spat3int.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=intrn --idattr=Parent --mode=union ./BAM_sorted/troc1s.bam ./features/intron.gff > troc1int.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=intrn --idattr=Parent --mode=union ./BAM_sorted/troc2s.bam ./features/intron.gff > troc2int.counts

# Count all reads in repeats:
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=similarity --idattr=Target --mode=union ./BAM_sorted/2nst0_1s.bam ./features/repeats.gff > 2nst0_1rep.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=similarity --idattr=Target --mode=union ./BAM_sorted/2nst0_2s.bam ./features/repeats.gff > 2nst0_2rep.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=similarity --idattr=Target --mode=union ./BAM_sorted/2nst0_3s.bam ./features/repeats.gff > 2nst0_3rep.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=similarity --idattr=Target --mode=union ./BAM_sorted/3nst0_1s.bam ./features/repeats.gff > 3nst0_1rep.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=similarity --idattr=Target --mode=union ./BAM_sorted/3nst0_2s.bam ./features/repeats.gff > 3nst0_2rep.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=similarity --idattr=Target --mode=union ./BAM_sorted/3nst0_3s.bam ./features/repeats.gff > 3nst0_3rep.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=similarity --idattr=Target --mode=union ./BAM_sorted/3nst3fa_1s.bam ./features/repeats.gff > 3nst3fa_1rep.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=similarity --idattr=Target --mode=union ./BAM_sorted/3nst3fa_2s.bam ./features/repeats.gff > 3nst3fa_2rep.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=similarity --idattr=Target --mode=union ./BAM_sorted/3nst3fa_3s.bam ./features/repeats.gff > 3nst3fa_3rep.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=similarity --idattr=Target --mode=union ./BAM_sorted/3nst3fb_1s.bam ./features/repeats.gff > 3nst3fb_1rep.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=similarity --idattr=Target --mode=union ./BAM_sorted/3nst3fb_2s.bam ./features/repeats.gff > 3nst3fb_2rep.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=similarity --idattr=Target --mode=union ./BAM_sorted/3nst3fb_3s.bam ./features/repeats.gff > 3nst3fb_3rep.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=similarity --idattr=Target --mode=union ./BAM_sorted/3nst23ma_1s.bam ./features/repeats.gff > 3nst23ma_1rep.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=similarity --idattr=Target --mode=union ./BAM_sorted/3nst23ma_2s.bam ./features/repeats.gff > 3nst23ma_2rep.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=similarity --idattr=Target --mode=union ./BAM_sorted/3nst23ma_3s.bam ./features/repeats.gff > 3nst23ma_3rep.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=similarity --idattr=Target --mode=union ./BAM_sorted/3nst23mb_1s.bam ./features/repeats.gff > 3nst23mb_1rep.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=similarity --idattr=Target --mode=union ./BAM_sorted/3nst23mb_2s.bam ./features/repeats.gff > 3nst23mb_2rep.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=similarity --idattr=Target --mode=union ./BAM_sorted/3nst23mb_3s.bam ./features/repeats.gff > 3nst23mb_3rep.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=similarity --idattr=Target --mode=union ./BAM_sorted/28cell1s.bam ./features/repeats.gff > 28cell1rep.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=similarity --idattr=Target --mode=union ./BAM_sorted/28cell2s.bam ./features/repeats.gff > 28cell2rep.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=similarity --idattr=Target --mode=union ./BAM_sorted/28cell3s.bam ./features/repeats.gff > 28cell3rep.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=similarity --idattr=Target --mode=union ./BAM_sorted/blastula2s.bam ./features/repeats.gff > blastula2rep.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=similarity --idattr=Target --mode=union ./BAM_sorted/blastula3s.bam ./features/repeats.gff > blastula3rep.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=similarity --idattr=Target --mode=union ./BAM_sorted/dlarv1s.bam ./features/repeats.gff > dlarv1rep.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=similarity --idattr=Target --mode=union ./BAM_sorted/dlarv2s.bam ./features/repeats.gff > dlarv2rep.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=similarity --idattr=Target --mode=union ./BAM_sorted/dlarv3s.bam ./features/repeats.gff > dlarv3rep.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=similarity --idattr=Target --mode=union ./BAM_sorted/gastrula1s.bam ./features/repeats.gff > gastrula1rep.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=similarity --idattr=Target --mode=union ./BAM_sorted/gastrula2s.bam ./features/repeats.gff > gastrula2rep.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=similarity --idattr=Target --mode=union ./BAM_sorted/gastrula3s.bam ./features/repeats.gff > gastrula3rep.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=similarity --idattr=Target --mode=union ./BAM_sorted/morula1s.bam ./features/repeats.gff > morula1rep.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=similarity --idattr=Target --mode=union ./BAM_sorted/morula2s.bam ./features/repeats.gff > morula2rep.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=similarity --idattr=Target --mode=union ./BAM_sorted/ovo1s.bam ./features/repeats.gff > ovo1rep.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=similarity --idattr=Target --mode=union ./BAM_sorted/ovo2s.bam ./features/repeats.gff > ovo2rep.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=similarity --idattr=Target --mode=union ./BAM_sorted/ovo3s.bam ./features/repeats.gff > ovo3rep.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=similarity --idattr=Target --mode=union ./BAM_sorted/spat1s.bam ./features/repeats.gff > spat1rep.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=similarity --idattr=Target --mode=union ./BAM_sorted/spat2s.bam ./features/repeats.gff > spat2rep.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=similarity --idattr=Target --mode=union ./BAM_sorted/spat3s.bam ./features/repeats.gff > spat3rep.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=similarity --idattr=Target --mode=union ./BAM_sorted/troc1s.bam ./features/repeats.gff > troc1rep.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=similarity --idattr=Target --mode=union ./BAM_sorted/troc2s.bam ./features/repeats.gff > troc2rep.counts

# Count all reads in 100 bp or greater promoter regions:
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=promoter --idattr=ID --mode=union ./BAM_sorted/2nst0_1s.bam ./features/promoter100bp.gff > 2nst0_1pro.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=promoter --idattr=ID --mode=union ./BAM_sorted/2nst0_2s.bam ./features/promoter100bp.gff > 2nst0_2pro.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=promoter --idattr=ID --mode=union ./BAM_sorted/2nst0_3s.bam ./features/promoter100bp.gff > 2nst0_3pro.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=promoter --idattr=ID --mode=union ./BAM_sorted/3nst0_1s.bam ./features/promoter100bp.gff > 3nst0_1pro.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=promoter --idattr=ID --mode=union ./BAM_sorted/3nst0_2s.bam ./features/promoter100bp.gff > 3nst0_2pro.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=promoter --idattr=ID --mode=union ./BAM_sorted/3nst0_3s.bam ./features/promoter100bp.gff > 3nst0_3pro.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=promoter --idattr=ID --mode=union ./BAM_sorted/3nst3fa_1s.bam ./features/promoter100bp.gff > 3nst3fa_1pro.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=promoter --idattr=ID --mode=union ./BAM_sorted/3nst3fa_2s.bam ./features/promoter100bp.gff > 3nst3fa_2pro.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=promoter --idattr=ID --mode=union ./BAM_sorted/3nst3fa_3s.bam ./features/promoter100bp.gff > 3nst3fa_3pro.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=promoter --idattr=ID --mode=union ./BAM_sorted/3nst3fb_1s.bam ./features/promoter100bp.gff > 3nst3fb_1pro.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=promoter --idattr=ID --mode=union ./BAM_sorted/3nst3fb_2s.bam ./features/promoter100bp.gff > 3nst3fb_2pro.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=promoter --idattr=ID --mode=union ./BAM_sorted/3nst3fb_3s.bam ./features/promoter100bp.gff > 3nst3fb_3pro.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=promoter --idattr=ID --mode=union ./BAM_sorted/3nst23ma_1s.bam ./features/promoter100bp.gff > 3nst23ma_1pro.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=promoter --idattr=ID --mode=union ./BAM_sorted/3nst23ma_2s.bam ./features/promoter100bp.gff > 3nst23ma_2pro.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=promoter --idattr=ID --mode=union ./BAM_sorted/3nst23ma_3s.bam ./features/promoter100bp.gff > 3nst23ma_3pro.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=promoter --idattr=ID --mode=union ./BAM_sorted/3nst23mb_1s.bam ./features/promoter100bp.gff > 3nst23mb_1pro.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=promoter --idattr=ID --mode=union ./BAM_sorted/3nst23mb_2s.bam ./features/promoter100bp.gff > 3nst23mb_2pro.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=promoter --idattr=ID --mode=union ./BAM_sorted/3nst23mb_3s.bam ./features/promoter100bp.gff > 3nst23mb_3pro.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=promoter --idattr=ID --mode=union ./BAM_sorted/28cell1s.bam ./features/promoter100bp.gff > 28cell1pro.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=promoter --idattr=ID --mode=union ./BAM_sorted/28cell2s.bam ./features/promoter100bp.gff > 28cell2pro.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=promoter --idattr=ID --mode=union ./BAM_sorted/28cell3s.bam ./features/promoter100bp.gff > 28cell3pro.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=promoter --idattr=ID --mode=union ./BAM_sorted/blastula2s.bam ./features/promoter100bp.gff > blastula2pro.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=promoter --idattr=ID --mode=union ./BAM_sorted/blastula3s.bam ./features/promoter100bp.gff > blastula3pro.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=promoter --idattr=ID --mode=union ./BAM_sorted/dlarv1s.bam ./features/promoter100bp.gff > dlarv1pro.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=promoter --idattr=ID --mode=union ./BAM_sorted/dlarv2s.bam ./features/promoter100bp.gff > dlarv2pro.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=promoter --idattr=ID --mode=union ./BAM_sorted/dlarv3s.bam ./features/promoter100bp.gff > dlarv3pro.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=promoter --idattr=ID --mode=union ./BAM_sorted/gastrula1s.bam ./features/promoter100bp.gff > gastrula1pro.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=promoter --idattr=ID --mode=union ./BAM_sorted/gastrula2s.bam ./features/promoter100bp.gff > gastrula2pro.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=promoter --idattr=ID --mode=union ./BAM_sorted/gastrula3s.bam ./features/promoter100bp.gff > gastrula3pro.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=promoter --idattr=ID --mode=union ./BAM_sorted/morula1s.bam ./features/promoter100bp.gff > morula1pro.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=promoter --idattr=ID --mode=union ./BAM_sorted/morula2s.bam ./features/promoter100bp.gff > morula2pro.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=promoter --idattr=ID --mode=union ./BAM_sorted/ovo1s.bam ./features/promoter100bp.gff > ovo1pro.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=promoter --idattr=ID --mode=union ./BAM_sorted/ovo2s.bam ./features/promoter100bp.gff > ovo2pro.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=promoter --idattr=ID --mode=union ./BAM_sorted/ovo3s.bam ./features/promoter100bp.gff > ovo3pro.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=promoter --idattr=ID --mode=union ./BAM_sorted/spat1s.bam ./features/promoter100bp.gff > spat1pro.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=promoter --idattr=ID --mode=union ./BAM_sorted/spat2s.bam ./features/promoter100bp.gff > spat2pro.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=promoter --idattr=ID --mode=union ./BAM_sorted/spat3s.bam ./features/promoter100bp.gff > spat3pro.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=promoter --idattr=ID --mode=union ./BAM_sorted/troc1s.bam ./features/promoter100bp.gff > troc1pro.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=promoter --idattr=ID --mode=union ./BAM_sorted/troc2s.bam ./features/promoter100bp.gff > troc2pro.counts

# Count all reads in transposable elements:
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=Tandem_Repeat --idattr=ID --mode=union ./BAM_sorted/2nst0_1s.bam ./features/teattr.gff > 2nst0_1te.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=Tandem_Repeat --idattr=ID --mode=union ./BAM_sorted/2nst0_2s.bam ./features/teattr.gff > 2nst0_2te.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=Tandem_Repeat --idattr=ID --mode=union ./BAM_sorted/2nst0_3s.bam ./features/teattr.gff > 2nst0_3te.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=Tandem_Repeat --idattr=ID --mode=union ./BAM_sorted/3nst0_1s.bam ./features/teattr.gff > 3nst0_1te.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=Tandem_Repeat --idattr=ID --mode=union ./BAM_sorted/3nst0_2s.bam ./features/teattr.gff > 3nst0_2te.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=Tandem_Repeat --idattr=ID --mode=union ./BAM_sorted/3nst0_3s.bam ./features/teattr.gff > 3nst0_3te.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=Tandem_Repeat --idattr=ID --mode=union ./BAM_sorted/3nst3fa_1s.bam ./features/teattr.gff > 3nst3fa_1te.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=Tandem_Repeat --idattr=ID --mode=union ./BAM_sorted/3nst3fa_2s.bam ./features/teattr.gff > 3nst3fa_2te.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=Tandem_Repeat --idattr=ID --mode=union ./BAM_sorted/3nst3fa_3s.bam ./features/teattr.gff > 3nst3fa_3te.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=Tandem_Repeat --idattr=ID --mode=union ./BAM_sorted/3nst3fb_1s.bam ./features/teattr.gff > 3nst3fb_1te.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=Tandem_Repeat --idattr=ID --mode=union ./BAM_sorted/3nst3fb_2s.bam ./features/teattr.gff > 3nst3fb_2te.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=Tandem_Repeat --idattr=ID --mode=union ./BAM_sorted/3nst3fb_3s.bam ./features/teattr.gff > 3nst3fb_3te.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=Tandem_Repeat --idattr=ID --mode=union ./BAM_sorted/3nst23ma_1s.bam ./features/teattr.gff > 3nst23ma_1te.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=Tandem_Repeat --idattr=ID --mode=union ./BAM_sorted/3nst23ma_2s.bam ./features/teattr.gff > 3nst23ma_2te.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=Tandem_Repeat --idattr=ID --mode=union ./BAM_sorted/3nst23ma_3s.bam ./features/teattr.gff > 3nst23ma_3te.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=Tandem_Repeat --idattr=ID --mode=union ./BAM_sorted/3nst23mb_1s.bam ./features/teattr.gff > 3nst23mb_1te.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=Tandem_Repeat --idattr=ID --mode=union ./BAM_sorted/3nst23mb_2s.bam ./features/teattr.gff > 3nst23mb_2te.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=Tandem_Repeat --idattr=ID --mode=union ./BAM_sorted/3nst23mb_3s.bam ./features/teattr.gff > 3nst23mb_3te.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=Tandem_Repeat --idattr=ID --mode=union ./BAM_sorted/28cell1s.bam ./features/teattr.gff > 28cell1te.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=Tandem_Repeat --idattr=ID --mode=union ./BAM_sorted/28cell2s.bam ./features/teattr.gff > 28cell2te.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=Tandem_Repeat --idattr=ID --mode=union ./BAM_sorted/28cell3s.bam ./features/teattr.gff > 28cell3te.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=Tandem_Repeat --idattr=ID --mode=union ./BAM_sorted/blastula2s.bam ./features/teattr.gff > blastula2te.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=Tandem_Repeat --idattr=ID --mode=union ./BAM_sorted/blastula3s.bam ./features/teattr.gff > blastula3te.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=Tandem_Repeat --idattr=ID --mode=union ./BAM_sorted/dlarv1s.bam ./features/teattr.gff > dlarv1te.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=Tandem_Repeat --idattr=ID --mode=union ./BAM_sorted/dlarv2s.bam ./features/teattr.gff > dlarv2te.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=Tandem_Repeat --idattr=ID --mode=union ./BAM_sorted/dlarv3s.bam ./features/teattr.gff > dlarv3te.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=Tandem_Repeat --idattr=ID --mode=union ./BAM_sorted/gastrula1s.bam ./features/teattr.gff > gastrula1te.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=Tandem_Repeat --idattr=ID --mode=union ./BAM_sorted/gastrula2s.bam ./features/teattr.gff > gastrula2te.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=Tandem_Repeat --idattr=ID --mode=union ./BAM_sorted/gastrula3s.bam ./features/teattr.gff > gastrula3te.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=Tandem_Repeat --idattr=ID --mode=union ./BAM_sorted/morula1s.bam ./features/teattr.gff > morula1te.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=Tandem_Repeat --idattr=ID --mode=union ./BAM_sorted/morula2s.bam ./features/teattr.gff > morula2te.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=Tandem_Repeat --idattr=ID --mode=union ./BAM_sorted/ovo1s.bam ./features/teattr.gff > ovo1te.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=Tandem_Repeat --idattr=ID --mode=union ./BAM_sorted/ovo2s.bam ./features/teattr.gff > ovo2te.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=Tandem_Repeat --idattr=ID --mode=union ./BAM_sorted/ovo3s.bam ./features/teattr.gff > ovo3te.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=Tandem_Repeat --idattr=ID --mode=union ./BAM_sorted/spat1s.bam ./features/teattr.gff > spat1te.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=Tandem_Repeat --idattr=ID --mode=union ./BAM_sorted/spat2s.bam ./features/teattr.gff > spat2te.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=Tandem_Repeat --idattr=ID --mode=union ./BAM_sorted/spat3s.bam ./features/teattr.gff > spat3te.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=Tandem_Repeat --idattr=ID --mode=union ./BAM_sorted/troc1s.bam ./features/teattr.gff > troc1te.counts
/usr/bin/python -m HTSeq.scripts.count --format=bam --stranded=no --type=Tandem_Repeat --idattr=ID --mode=union ./BAM_sorted/troc2s.bam ./features/teattr.gff > troc2te.counts



